exonum-java-binding/core/rust/src/utils/errors.rs

```diff
@@ -70,6 +70,30 @@ pub fn check_error_on_exception<T>(env: &JNIEnv, result: JniResult<T>) -> Result
     })
 }
 
+/// Handles and clears any Java exceptions.
+///
+/// Any JNI errors are logged with their descriptions,  for JNI errors
+/// like `JniErrorKind::JavaException` it gets (and clears) any exception
+/// that is currently being thrown.
+///
+/// Panics:
+/// - Panics if JNI error is `JniErrorKind::JavaException` but couldn't get (and clear) exception
+///   object and describe it.
+pub fn log_jni_error_or_exception<T>(env: &JNIEnv, result: JniResult<T>) -> JniResult<T> {
```

Comment:
I'd consider renaming to communicate that it not just logs (it is a secondary responsibility, but handles/clears the exception, if any).
            handle_exception/clear_exception?